src/app/customer-search/customer-search.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit, Injector, ComponentFactoryResolver, ApplicationRef } from '@angular/core';
+import { TestsDataService } from '../services/tests-data.service';
+import { TsScreenerDataService } from '../services/ts-screener-data.service';
+import { ThsDataService } from '../services/ths-data.service';
+import { TfiDataService } from '../services/tfi-data.service';
+
+@Component({
+  selector: 'customer-search',
+  templateUrl: './customer-search.component.html',
+  styleUrls: ['./customer-search.component.css']
+})
+export class CustomerSearchComponent implements OnInit {
+  public idSearch: string;
+  public searchBtn: boolean = true; // Search button is disabled while querying DB
+  public resultsTable = [];
+  public currentPage: number = 0;
+
+  constructor(
+    private testDataService: TestsDataService, private tsDataService: TsScreenerDataService, private thsDataService: ThsDataService, private tfiDataService: TfiDataService)
+    { }
+
+  ngOnInit() {
+  }
+
+  // Gets JSON from queryDB() puts it into an array to create the table of appointments
+  // sorts by date, most recent first.
+  public patientSearch() {
+    this.searchBtn = false;
+    let appts: Object[] = this.queryDB();
+    this.currentPage = 0;
+    this.resultsTable = appts.map(x => Object.assign({}, x));
+    this.resultsTable.sort((a,b) => {
+      let date1 = new Date(a.date);
+      let date2 = new Date(b.date);
+      if(date1.valueOf() < date2.valueOf()) {
+        return 1;
+      }
+      return -1;
+    });
+    this.searchBtn = true;
+  }
+  // CHANGE this function to actually load the selected appointment into sessionStorage and tell
+  // audiologist-navigation to change state
+  public loadAppt(appt: Object) {
+    if(appt === {}) { return;}
```

Comment:
This condition is probably false more often than you think. Instead, check whether the properties you need are undefined.